tests/endtoend_tests/test_endtoend.py

```
@@ -128,6 +129,54 @@ def test_build_generate_only(dummy_tesseract_location, skip_checks):
             assert 'RUN ["tesseract-runtime", "check"]' in docker_file_contents
 
 
+def test_env_passthrough_serve(
+    docker_cleanup, docker_client, dummy_image_name, dummy_tesseract_location
+):
+    """Ensure we can pass environment variables to tesseracts when serving."""
+    image_name = build_tesseract(
+        docker_client,
+        dummy_tesseract_location,
+        dummy_image_name,
+    )
+    assert image_exists(docker_client, image_name)
+    docker_cleanup["images"].append(image_name)
+
+    cli_runner = CliRunner(mix_stderr=False)
+    run_res = subprocess.run(
+        [
+            "tesseract",
+            "serve",
+            image_name,
+            "--env=TEST_ENV_VAR=foo",
+        ],
+        capture_output=True,
+        cwd=dummy_tesseract_location,
+        text=True,
+    )
+    assert run_res.returncode == 0, run_res.stderr
+    assert run_res.stdout
+
+    project_meta = json.loads(run_res.stdout)
+    project_id = project_meta["project_id"]
+    tesseract_id = project_meta["containers"][0]["name"]
+
+    try:
+        container = docker_client.containers.get(tesseract_id)
+        exit_code, output = container.exec_run(["sh", "-c", "echo $TEST_ENV_VAR"])
+        assert exit_code == 0, f"Command failed with exit code {exit_code}"
+        assert "foo" in output.decode("utf-8"), f"Output was: {output.decode('utf-8')}"
+    except ContainerError as e:
+        pytest.fail(f"Failed to execute command in container: {e}")
```

Why this branch? I'd just let it bubble up.